### PR TITLE
adding sign up page + input validation

### PR DESCRIPTION
**Pull request by tanya-raj**

### CODE REVIEW

File: react-site/src/pages/SignInPage/SignInPage.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React from "react";
+import { useHistory } from "react-router-dom";
 
 import "./SignInPage.css";
+import SignUpPage from "../SignUpPage";
```
Comment (svasandani) [inline react-site/src/pages/SignInPage/SignInPage.js:5]: Do we need to import this here?

**Comment by svasandani** [comment]
Screenshot_20211212-011332

Small issue on mobile, card overflows into nav. Setting min height of the page might resolve this or adding padding

File: react-site/src/pages/SignUpPage/SignUpPage.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        variant="default"
+        color="white"
+        style={{
+          padding: "36px",
+          zIndex: "1",
+          maxWidth: "90vw",
+        }}
+      >
+        <Heading>new here?</Heading>
+        <Text>clear skies are only a few taps away.</Text>
```
Comment (rebeccashi) [inline react-site/src/pages/SignUpPage/SignUpPage.js:39]: Can we make this less abstract haha

File: react-site/src/pages/SignUpPage/SignUpPage.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            placeholder="email"
+            withIcon={false}
+            label="email"
+            value={email}
+            setValue={setEmail}
+            style={{
+              width: "100%",
+            }}
+            type="email"
+            error={
```
Comment (rebeccashi) [inline react-site/src/pages/SignUpPage/SignUpPage.js:62]: I think we should also check the password length to be at least 8 on the frontend, so we don't let them submit to backend and get a "bad request" response

File: react-site/src/pages/SignUpPage/SignUpPage.js (hunk 2 above)
Comment (rebeccashi) [reply react-site/src/pages/SignUpPage/SignUpPage.js:39]: I'm talking about "clear skies are only a few taps away."

**Review by rebeccashi — APPROVED** [review]

**Comment by rebeccashi** [comment]
The options in visualizations are cool! Honestly we should possibly consider this as a premium feature for members only.

**Pull request merged**